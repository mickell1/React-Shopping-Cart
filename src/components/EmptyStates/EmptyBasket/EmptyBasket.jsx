import React from "react";
import emptybasket from '../../../images/empty-basket.png';
import './EmptyBasket.scss';

const EmptyBasket = () => {
  return (
    <div className="empty-basket">
      <img
        src={emptybasket}
        alt="empty-basket"
      />
      <h2>Your basket is empty!</h2>
    </div>
  );
};

export default EmptyBasket;
