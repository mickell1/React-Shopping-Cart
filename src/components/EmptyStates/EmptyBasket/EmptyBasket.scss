@import '../../../variables.scss';

.empty-basket {
  text-align: center;
  img {
    width: 100%;
    max-width: 250px;
    margin: 42px auto 16px auto;
  }
  h2 {
    color: $color-gray-light;
    font-size: 18px;
  }
}