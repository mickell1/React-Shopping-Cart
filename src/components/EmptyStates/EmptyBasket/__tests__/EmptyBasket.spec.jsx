import React from 'react';
import { shallow, mount } from 'enzyme';
import EmptyBasket from '../EmptyBasket';

describe('EmptyBasket', () => {

  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<EmptyBasket /> )  
  })

  it('should render the EmptyBasket Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });

  it('should render classname', () => {
    wrapper = mount(
     <EmptyBasket />
   );

   expect(wrapper.find('div').first().hasClass('empty-basket')).toEqual(true);
 });

  it('should render h2', () => {
    wrapper = mount(<EmptyBasket />)
    expect(wrapper.find('h2').text()).toBe('Your basket is empty!')
  })
});