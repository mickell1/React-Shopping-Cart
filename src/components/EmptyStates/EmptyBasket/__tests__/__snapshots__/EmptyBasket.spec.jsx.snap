// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyBasket should render the EmptyBasket Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyBasket />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="empty-basket"
          src="empty-basket.png"
        />,
        <h2>
          Your basket is empty!
        </h2>,
      ],
      "className": "empty-basket",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "empty-basket",
          "src": "empty-basket.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your basket is empty!",
        },
        "ref": null,
        "rendered": "Your basket is empty!",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="empty-basket"
            src="empty-basket.png"
          />,
          <h2>
            Your basket is empty!
          </h2>,
        ],
        "className": "empty-basket",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "empty-basket",
            "src": "empty-basket.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your basket is empty!",
          },
          "ref": null,
          "rendered": "Your basket is empty!",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
